airflow/models.py

```diff
@@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
             DR.dag_id == self.dag_id,
             DR.execution_date == self.execution_date,
             DR.run_id == self.run_id
-        ).one()
+        ).first()
```

Comment:
Both will attempt to return a single query result row, but .one() will
throw an exception if there are no entries whereas .first() safely returns
None.

On Wed, Sep 7, 2016 at 10:25 PM, bolkedebruin <notifications@github.com>
wrote:

> In airflow/models.py
> <https://github.com/apache/incubator-airflow/pull/1730#discussion_r77947623>
> :
>
> > @@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
> >              DR.dag_id == self.dag_id,
> >              DR.execution_date == self.execution_date,
> >              DR.run_id == self.run_id
> > -        ).one()
> > +        ).first()
>
> why the change to first? this changes the contract of the function?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-airflow/pull/1730/files/a03fe0478cf7627255782696f083de4deeb8640e#r77947623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGYMpsrJ-qNrpkQrHeVRFnzWigPYfYGUks5qn5xPgaJpZM4JjjUq>
> .
>



-- 
Thank you,

Norman Mu